platforms/android/src/jp/wizcorp/phonegap/plugin/WizAssets/WizAssetsPlugin.java

```diff
@@ -317,80 +330,147 @@ public AsyncDownload(String url, String uri, String filePath, CallbackContext ca
+                        return null;
+                    }
+                    int statusCode = statusLine.getStatusCode();
+                    if (statusCode < 200 || statusCode > 299) {
+                        this.callbackContext.error(createDownloadFileError(HTTP_REQUEST_ERROR, statusCode));
+                        return null;
+                    }
+                } catch (ClientProtocolException e) {
+                    this.callbackContext.error(createDownloadFileError(CONNECTIVITY_ERROR));
+                    return null;
+                } catch (IOException e) {
+                    this.callbackContext.error(createDownloadFileError(CONNECTIVITY_ERROR));
+                    return null;
                 }
 
-                HttpResponse response = httpclient.execute(httpRequest);
                 HttpEntity entity = response.getEntity();
 
                 InputStream is;
 
                 Header contentHeader = entity.getContentEncoding();
-                if (contentHeader != null) {
-                    if (contentHeader.getValue().contains("gzip")) {
-                        Log.d(TAG, "GGGGGGGGGZIIIIIPPPPPED!");
+                try {
+                    String contentHeaderValue = contentHeader != null ? contentHeader.getValue() : null;
+                    if (contentHeaderValue != null && contentHeaderValue.contains("gzip")) {
                         is = new GZIPInputStream(entity.getContent());
                     } else {
                         BufferedHttpEntity bufHttpEntity = new BufferedHttpEntity(entity);
                         is = bufHttpEntity.getContent();
                     }
-                } else {
-                    BufferedHttpEntity bufHttpEntity = new BufferedHttpEntity(entity);
-                    is = bufHttpEntity.getContent();
+                    if (is == null) {
+                        this.callbackContext.error(createDownloadFileError(HTTP_REQUEST_CONTENT_ERROR));
+                        return null;
+                    }
+                } catch (IOException e) {
+                    this.callbackContext.error(createDownloadFileError(HTTP_REQUEST_CONTENT_ERROR));
+                    return null;
                 }
+
                 byte[] buffer = new byte[1024];
 
                 int len1 = 0;
 
-                FileOutputStream fos = new FileOutputStream(file);
+                FileOutputStream fos;
+                try {
+                    fos = new FileOutputStream(file);
 
-                while ((len1 = is.read(buffer)) > 0 ) {
-                    fos.write(buffer,0, len1);
-                }
+                    while ((len1 = is.read(buffer)) > 0 ) {
+                        fos.write(buffer, 0, len1);
+                    }
 
-                fos.close();
-                is.close();
+                    fos.close();
+                    is.close();
```

Comment:
If an error occurs during read or write, these file descriptors will never be closed. They're better off in a `finally` block.